app/src/main/java/com/sci/recipeandroid/feature/auth/ui/screen/authentication/AuthOptionFragment.kt

```diff
@@ -0,0 +1,129 @@
+package com.sci.recipeandroid.feature.auth.ui.screen.authentication
+
+import android.content.Intent
+import android.graphics.Typeface
+import android.os.Bundle
+import android.text.SpannableString
+import android.text.Spanned
+import android.text.style.StyleSpan
+import android.text.style.UnderlineSpan
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.updateLayoutParams
+import androidx.fragment.app.Fragment
+import com.facebook.AccessToken
+import com.facebook.CallbackManager
+import com.facebook.FacebookCallback
+import com.facebook.FacebookException
+import com.sci.recipeandroid.R
+import com.sci.recipeandroid.databinding.FragmentAuthenticationOptionsBinding
+import com.sci.recipeandroid.feature.auth.ui.viewmodel.AuthOptionViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+import com.facebook.FacebookSdk;
+import com.facebook.appevents.AppEventsLogger;
+import com.facebook.login.LoginManager
+import com.facebook.login.LoginResult
+import com.sci.recipeandroid.MainActivity
+import com.sci.recipeandroid.util.setOneTimeClickListener
+
+
+class AuthOptionFragment : Fragment() {
+
+    private var _binding: FragmentAuthenticationOptionsBinding? = null
+    private val binding get() = _binding!!
+
+    private val authViewModel: AuthOptionViewModel by viewModel()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentAuthenticationOptionsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        authViewModel.uiEvent.observe(viewLifecycleOwner) {
+            Toast.makeText(this.context, it, Toast.LENGTH_LONG).show()
+        }
+        binding.googleBtn.setOneTimeClickListener {
+            authViewModel.googleAuthentication(requireActivity())
```

Comment:
plz inject context from koin to dependency  client class

Reply:
inject this context using the custom context 9baf80468f57d78d6c4b13f545e1ebaa10f9d59c